banyand/webapp/webapp.go

```diff
@@ -0,0 +1,153 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package webapp
+
+import (
+	"fmt"
+	"io/fs"
+	"net/http"
+	"strings"
+
+	"github.com/apache/skywalking-banyandb/pkg/logger"
+	"github.com/apache/skywalking-banyandb/pkg/run"
+	"github.com/apache/skywalking-banyandb/ui"
+)
+
+type ServiceRepo interface {
+	run.Config
+	run.Service
+}
+
+var (
+	_ ServiceRepo = (*service)(nil)
+)
+
+func NewService() ServiceRepo {
+	return &service{
+		stopCh: make(chan struct{}),
+	}
+}
+
+type service struct {
+	listenAddr string
+	mux        *http.ServeMux
+	stopCh     chan struct{}
+	l          *logger.Logger
+}
+
+func (p *service) FlagSet() *run.FlagSet {
+	flagSet := run.NewFlagSet("")
+	flagSet.StringVar(&p.listenAddr, "webapp-listener-addr", ":17913", "listen addr for webapp")
```

Comment:
We will server gRPC-web protocol on this port or `17912`?

Reply:
I prefer this port. The frontend could access API through `/api/*`

Reply:
I agree. Use one protocol for a single port may be the best practice.